test/functional/specs/LibraryInfo/C2589.js

```diff
@@ -49,9 +49,16 @@ test("C2589: getLibraryInfo command returns library information.", async () => {
 
   const alloy = createAlloyProxy();
   await alloy.configure(debugEnabledConfig);
-  const libraryInfo = await alloy.getLibraryInfo();
-  delete libraryInfo.libraryInfo.configs.edgeBasePath;
-  await t.expect(libraryInfo.libraryInfo.version).eql(currentVersion);
-  await t.expect(libraryInfo.libraryInfo.commands).eql(currentCommand);
-  await t.expect(libraryInfo.libraryInfo.configs).eql(currentConfigs);
+  const { libraryInfo } = await alloy.getLibraryInfo();
+  delete libraryInfo.configs.edgeBasePath;
+  await t.expect(libraryInfo.version).eql(currentVersion);
+  await t.expect(libraryInfo.commands).eql(currentCommand);
+  await t.expect(libraryInfo.configs).eql(currentConfigs);
+});
+
+test("C2589: getLibraryInfo correctly serializes functions in the config", async () => {
+  const alloy = createAlloyProxy();
+  await alloy.configure(debugEnabledConfig);
+  const { libraryInfo } = await alloy.getLibraryInfo();
+  await t.expect(typeof libraryInfo.configs.onBeforeEventSend).eql("string");
```

Comment:
Let's try to catch future errors too. How about "libraryInfo can be marshaled to postMessage". Also, once the value is marshaled into the testCafe node environment we are testing the testCafe marshaling system instead of the browser.

I think you'd want to create a client function that calls getLibraryInfo, and then tries to call window.postMessage with the response.